File: algorithms/easy/find_the_array_concatenation_value.cpp
class Solution {
public:
  long long findTheArrayConcVal(vector<int> &nums) {
    long long ans = 0;
    for (int i = 0, j = nums.size() - 1; i <= j; i++, j--) {
      string s = to_string(nums[i]);
      if (i < j)
        s += to_string(nums[j]);
      ans += stoll(s, nullptr, 10);
    }
    return ans;
  }
};

File: algorithms/easy/find_the_array_concatenation_value.ex
defmodule Solution do
  @spec find_the_array_conc_val(nums :: [integer]) :: integer
  def find_the_array_conc_val(nums) do
    len = length(nums) |> div(2)
    {f, r} = nums |> Enum.split(len)
    aux(f, Enum.reverse(r), 0)
  end

  def aux([], [], ans), do: ans
  def aux([], [y], ans), do: ans + y

  def aux([x | xs], [y | ys], ans) do
    aux(xs, ys, ans + String.to_integer("#{x}#{y}"))
  end
end
